flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSourceBuilder.java

```diff
@@ -369,6 +373,18 @@ public <T extends OUT> PulsarSourceBuilder<T> setDeserializationSchema(
         return self;
     }
 
+    /**
+     * Sets a {@link CryptoKeyReader}. Configure the key reader to be used to decrypt the message
+     * payloads.
+     *
+     * @param cryptoKeyReader CryptoKeyReader object
+     * @return this PulsarSourceBuilder.
+     */
+    public PulsarSourceBuilder<OUT> setCryptoKeyReader(CryptoKeyReader cryptoKeyReader) {
+        this.cryptoKeyReader = checkNotNull(cryptoKeyReader);
```

Comment:
Do we need to add e2e encryption in the documentation as well ? The cryptoKeyReader contains private key, I think we need to let users know what is the recommended/safe way to create a cryptoKeyReader~